### ISSUE TITLE
hevm: possible inaccuracy in gas readouts

### ISSUE THREAD

**Issue by realisation** [dapphub/dapptools#277]
I have a contract with a method that deploys another contract and calling this method costs me 5 millionish gas even though the other contract is very simple (inherits from a 3 small open zepellin ERC20 contracts). 

It deploys it via `MyContract myContract = new MyContract(arrayOf2to20Addresses);` and that contract writes the array to storage.

When I deploy that contract on Ropsten with Remix it only costs 1.5 million gas.

I'd report what it costs to call the method on my main contract to deploy it but I have to rewrite it to make it an appropriate length first.

**Comment by realisation** [comment]
The deployed bytecode of the contract being deployed is 7169, so that should be about 1433800 gas plus the opcodes' cost to be deployed from a contract, right?

**Comment by realisation** [comment]
Deploying the contract by itself that the factory contract was deploying reads out to 3041261 gas from hevm.

**Comment by MrChico** [comment]
Gas accounting should be pretty accurate now as we are passing the GeneralStateTests. If you can provide an isolated example we can look at it. Sounds like you might be looking at the cost for deploying both contracts

**Comment by realisation** [comment]
@MrChico @livnev 

I just tested again with a completely simplified version of the contract and see the same behaviors. To make it easy for you guys I swapped out the open zeppelin ERC20 I was using and used DSToken instead. 

Also here is a gist with the relevant files with remapped paths so you can copy/paste into remix for verification: https://gist.github.com/realisation/bab71e261ba847e21aaeaabbca569ff6
Just copy/paste, compile "sandbox", deploy it (you have to make sure the sandbox contract is selected when you deploy it - it will give a dropdown containing every contract from the inheritance tree), and run "create".


Using dapp test, the gas readout is 5708669 gas.

Using remix on Rinkeby and Kovan, it is 1910260 gas.

Here's the sandbox contract so you don't have to sort through that gist above if you just trust my remix figures:  

```
pragma solidity ^0.5.6;

import "../ERC20Token.sol";

contract Sandbox {

    ERC20Token token;

    function create () public {
        token = new ERC20Token("test a", "TESTA", 18, 100 * (10 ** 18));
    }

}
```

**Comment by MrChico** [comment]
Closed via #294
